gno.land/pkg/gnoclient/client_txs.go

```diff
@@ -33,63 +67,64 @@ type RunCfg struct {
 }
 
 // Call executes a contract call on the blockchain.
-func (c *Client) Call(cfg CallCfg) (*ctypes.ResultBroadcastTxCommit, error) {
+func (c *Client) Call(cfg BaseTxCfg, msgs ...MsgCall) (*ctypes.ResultBroadcastTxCommit, error) {
 	// Validate required client fields.
 	if err := c.validateSigner(); err != nil {
-		return nil, errors.Wrap(err, "validate signer")
+		return nil, err
 	}
 	if err := c.validateRPCClient(); err != nil {
-		return nil, errors.Wrap(err, "validate RPC client")
+		return nil, err
 	}
 
-	pkgPath := cfg.PkgPath
-	funcName := cfg.FuncName
-	args := cfg.Args
-	gasWanted := cfg.GasWanted
-	gasFee := cfg.GasFee
-	send := cfg.Send
-	sequenceNumber := cfg.SequenceNumber
-	accountNumber := cfg.AccountNumber
-	memo := cfg.Memo
-
-	// Validate config.
-	if pkgPath == "" {
-		return nil, errors.New("missing PkgPath")
+	// Validate base transaction config
+	if err := cfg.validateBaseTxConfig(); err != nil {
+		return nil, err
 	}
-	if funcName == "" {
-		return nil, errors.New("missing FuncName")
+
+	// Parse MsgCall slice
+	vmMsgs := make([]vm.MsgCall, len(msgs))
+	for _, msg := range msgs {
+		// Validate MsgCall fields
+		if err := msg.validateMsgCall(); err != nil {
+			return nil, err
+		}
+
+		// Parse send coins
+		send, err := std.ParseCoins(msg.Send)
+		if err != nil {
+			return nil, fmt.Errorf("%w", err)
+		}
+
+		// Unwrap syntax sugar to vm.MsgCall slice
+		vmMsgs = append(vmMsgs, vm.MsgCall{
+			Caller:  c.Signer.Info().GetAddress(),
+			PkgPath: msg.PkgPath,
+			Func:    msg.FuncName,
+			Send:    send,
+		})
 	}
 
-	// Parse send amount.
-	sendCoins, err := std.ParseCoins(send)
-	if err != nil {
-		return nil, errors.Wrap(err, "parsing send coins")
+	// Cast vm.MsgCall back into std.Msg
+	stdMsgs := make([]std.Msg, len(vmMsgs))
+	for i, msg := range vmMsgs {
+		stdMsgs[i] = msg
 	}
```

Comment:
I'm sure you can implement the same logic without needing to go back and forth on struct types.

Reply:
Hey, the issue is the fact that the `std.Tx` type has to take a in `std.Msg` slice. I also made the MsgCall wrapper in order to have a nice interface for developers, ie not to have them touch anything directly related to the VM.

Reply:
clarification: there is no way to avoid these two iterations because std.Msg is an interface...